registry.go

```diff
@@ -0,0 +1,114 @@
+package manet
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	ma "github.com/jbenet/go-multiaddr"
+)
+
+type AddrParser func(a net.Addr) (ma.Multiaddr, error)
+type MaddrParser func(ma ma.Multiaddr) (net.Addr, error)
+
+var maddrParsers map[string]MaddrParser
+var addrParsers map[string]AddrParser
+var addrParsersLock sync.Mutex
+
+type AddressSpec struct {
+	// NetNames is an array of strings that may be returned
+	// by net.Addr.Network() calls on addresses belonging to this type
+	NetNames []string
+
+	// Key is the string value for Multiaddr address keys
+	Key string
```

Comment:
oh i see. it's not called a "Key" nor "address keys". it's called a "Protocol Name". https://github.com/jbenet/multiaddr/blob/master/protocols.csv and https://github.com/jbenet/go-multiaddr/blob/master/protocols.go#L13

call this property `ProtocolName` or `Protocol`